### PR TITLE
MON-3867: generate CRD's and api

### PR DESCRIPTION
**Pull request by marioferh**
First PR to implement CRD Based CMO:

Add controller-gen, manifests and generate-crd to makefile
Add pkg/apis/cmo/v1
Add Generate CRD
Add example CustomResource

### CODE REVIEW

File: config/examples/clustermonitoringoperator.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+apiVersion: cmo.example.com/v1
```
Comment (marioferh) [inline config/examples/clustermonitoringoperator.yaml:1]: I need the correct name for this, some ideas?

Comment (juzhao) [reply config/examples/clustermonitoringoperator.yaml:1]: how about `monitoring.coreos.com/v1`

```
$  oc explain Prometheus
GROUP:      monitoring.coreos.com
KIND:       Prometheus
VERSION:    v1
...
```

Comment (marioferh) [reply config/examples/clustermonitoringoperator.yaml:1]: thanks, I don't know if correct group is that or cmo should go in another one

**Comment by marioferh** [comment]
moved to https://github.com/openshift/api/

**Pull request closed without merging**